frontend/workflows/projectCatalog/src/details/card.tsx

```diff
@@ -0,0 +1,149 @@
+import React from "react";
+import { Card, ClutchError, Error, Grid, styled, Typography } from "@clutch-sh/core";
+import { LinearProgress } from "@material-ui/core";
+
+export type CardTypes = "Dynamic" | "Metadata";
+
+export interface DetailsCard {
```

Comment:
nit: this name is a bit confusing. Just to confirm, this is solely used to identify the card type right?

Reply:
Yeah, it is a bit confusing. Wanted an interface that could be used as the child type interface which didn't interfere with the BaseCard. (even writing that was confusing lol).

Ultimately:

BaseCard inherits the type from DetailsCard (which is exported) via CardProps and BaseCardProps. 
The DynamicCard and MetaCard utilize the BaseCardProps to pass to the BaseCard and their individual types. 
Then any user can extend the DetailsCard to make a custom card with the required typing.

I think I ran out of names..